src/urlMatcherFactory.js

```diff
@@ -920,7 +924,12 @@ function $UrlMatcherFactory() {
       if (config.type && urlType) throw new Error("Param '"+id+"' has two type configurations.");
       if (urlType) return urlType;
       if (!config.type) return (location === "config" ? $types.any : $types.string);
```

Comment:
Heres where we were applying default type 'string' to url params (Actually, to non-non-url params ;) )